### PR TITLE
Skip items on trakt Exceptions

### PR DESCRIPTION
**Pull request by glensc**
refs:
- https://github.com/Taxel/PlexTraktSync/issues/270

### CODE REVIEW

**Comment by glensc** [comment]
@zerosquared can you test this and show what Trakt item is failing. maybe could have a nicer error message for this.

**Comment by zerosquared** [comment]
Thanks for the quick reply.  Now, I'm getting this...

Traceback (most recent call last):
  File "main.py", line 3, in <module>
    from plex_trakt_sync.cli import cli
  File "C:\Users\xxxs\Documents\GitHub\PlexTraktSync-main\plex_trakt_sync\cli.py", line 3, in <module>
    from plex_trakt_sync.commands.clear_collections import clear_collections
  File "C:\Users\xxxs\Documents\GitHub\PlexTraktSync-main\plex_trakt_sync\commands\clear_collections.py", line 2, in <module>
    from plex_trakt_sync.logging import logger
  File "C:\Users\xxxs\Documents\GitHub\PlexTraktSync-main\plex_trakt_sync\logging.py", line 30, in <module>
    initialize()
  File "C:\Users\xxxs\Documents\GitHub\PlexTraktSync-main\plex_trakt_sync\logging.py", line 9, in initialize
    log_level = logging.DEBUG if CONFIG['log_debug_messages'] else logging.INFO
  File "C:\Users\xxxs\Documents\GitHub\PlexTraktSync-main\plex_trakt_sync\config.py", line 26, in __getitem__
    self.initialize()
  File "C:\Users\xxxs\Documents\GitHub\PlexTraktSync-main\plex_trakt_sync\config.py", line 39, in initialize
    config = json.load(fp)
  File "C:\Program Files\Python38\lib\json\__init__.py", line 293, in load
    return loads(fp.read(),
  File "C:\Program Files\Python38\lib\json\__init__.py", line 357, in loads
    return _default_decoder.decode(s)
  File "C:\Program Files\Python38\lib\json\decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "C:\Program Files\Python38\lib\json\decoder.py", line 353, in raw_decode
    obj, end = self.scan_once(s, idx)
json.decoder.JSONDecodeError: Expecting ',' delimiter: line 20 column 3 (char 402)

**Comment by glensc** [comment]
@zerosquared your error seems to come from the clear collections command? please show the command line that you execute!

**Comment by glensc** [comment]
looks like you corrupted your `config.json`, it has a trailing comma on line 20, column 3

**Comment by zerosquared** [comment]
I found the trailing comma (in Exceptions) - fixed - running again.  Stand by for results. :)

**Comment by zerosquared** [comment]
I'm immediately getting this with the new sync.py file.

C:\Users\xxx\Documents\GitHub\PlexTraktSync-main>python main.py
Traceback (most recent call last):
  File "main.py", line 3, in <module>
    from plex_trakt_sync.cli import cli
  File "C:\Users\xxx\Documents\GitHub\PlexTraktSync-main\plex_trakt_sync\cli.py", line 6, in <module>
    from plex_trakt_sync.commands.sync import sync
  File "C:\Users\xxx\Documents\GitHub\PlexTraktSync-main\plex_trakt_sync\commands\sync.py", line 13, in <module>
    from plex_trakt_sync.version import git_version_info
ModuleNotFoundError: No module named 'plex_trakt_sync.version'

**Comment by zerosquared** [comment]
SO, I went back to the original scan.py file and tried again after I went through and started with the last successful add in my Movies collection and Refreshed the metadata on everything below that (W-Z) and now it's past that library and into the others.

I finally saw a Rate Limit Exceeded for Trakt.API, but after the timer was done, it continued on.

Script finally SUCCEEDED in the end.

Thank you for trying to help @glensc !

**Comment by glensc** [comment]
how was this error fixed?

```
ModuleNotFoundError: No module named 'plex_trakt_sync.version'
```

**Comment by zerosquared** [comment]
As stated, I went through and started with the last successful add that showed in the "last_update.log" and in my Movies collection in Plex, I did a "Refresh all Metadata" on everything below that (W-Z) in my Movies library.  I guess Trakt, or the script, didn't like one of my movies meta and made it crash?  Not sure, but once I refreshed the metadata and re-ran the original scan.py, it went all the way through successfully.

TLDR: Refresh your metadata where the crash happens and re-run script.  Repeat for any other conflicts until the script ends.

**Comment by glensc** [comment]
What is scan.py? What means you went through? The error you were getting was from trakt server, a simple retry would had helped with the new code. I added specially more info to the error log. And that you got the error again, I asked you to post it here. Did you even run with the code from this branch?

**Comment by zerosquared** [comment]
What?  "PlexTraktSync-main\plex_trakt_sync\commands\scan.py" is the name of the file that you made a change to and then asked me to try out.  It immediately gave me an error message.  See above comment "I'm immediately getting this with the new sync.py file."

So, while waiting for a fix, I clicked on "Refresh all Metadata" IN PLEX, on my Movies library, and after updating my movies, tried rerunning the script with the original files and it completed.

Not sure what else to tell you.  It worked after I updated my metadata in my movies library.
Thank you.

**Comment by glensc** [comment]
@zerosquared 
1. there's no scan.py in this project.
2. there's nothing wrong with this branch. the "ModuleNotFoundError" error you got does not happen in this branch, that's why I asked how you managed to get it. it's you doing something odd, and there's no fix coming from developers fixing your invalid code checkout errors.

seems you didn't even check out this branch and yet blame developers.

**Comment by glensc** [comment]
The original bug reporter did not provide the details asked, so going to merge as is.

**Comment by zerosquared** [comment]
Yeah, it was Sync.py all along

But, I never came here blaming anyone for anything.  I came in with an error message block that I didn't understand, you provided a fix, it didn't work (because I had left a trailing comma in the config file that you pointed out, I fixed it), I read some of the other comments about metadata so I refreshed my library and re-ran the original project files and everything worked.  

I even thanked you at the end and then you come back and chastise me for improper reporting, not using your branch to fix the issue, mis-typing a filename....I mean, you only changed ONE file in that branch to try and help and even though I was saying scan.py, you should've known it was sync.py that I meant.  You could've politely asked.

Never mind, man.  Again, thank  you for your project - it works - but you should be more considerate of your users when they have issues and be happy for them when they finally get it working.

Good luck.

(btw - your script hates TMDB ratings, at least on my server.  I refreshed all my movies to Rotten Tomatoes and IMDB ratings and that's when they processed through)

**Comment by glensc** [comment]
@zerosquared even the PR changed one file, your base branch was totally off, like 90+ commits! and I wanted you to report an error you were having with this branch, but you just discarded the request because it works for you. and you never said you copied only one file, I had to guess this from your responses.
